File: Data Structures & Algorithms/two-integer-sum/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int tg) {
        unordered_map<int, int> mp;
        vector<int> ans;
        mp[nums[0]]=0;
        for(int i=1; i<nums.size(); i++){
            int x = nums[i];
            if((nums[mp[tg-x]] + x) == tg) return {mp[tg-x], i};

            mp[x] = i;
        }
        return ans;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int ans = 0, cnt = 1;
        if (nums.length <= 1)
            return nums.length;

        

        Arrays.sort(nums);
        int last_smaller = nums[0];
        // System.out.println(Arrays.toString(nums));

        for (int i = 1; i < nums.length; i++) {
            // System.out.println("loop -> " + nums[i] + " ls -> "+ last_smaller);
            if (nums[i] - 1 == last_smaller) {
                cnt += 1;
                last_smaller = nums[i];
                // System.out.println(nums[i]);
            } else if (nums[i] != last_smaller) {
                cnt = 1;
                last_smaller = nums[i];
            }
            ans = Math.max(ans, cnt);
        }
        return ans;
    }
}
